cbmc/proofs/IotMqtt_Cleanup/IotMqtt_Cleanup_harness.c

```diff
@@ -0,0 +1,14 @@
+#include "iot_config.h"
+#include "private/iot_mqtt_internal.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+void harness()
+{
+  IotMqtt_Cleanup();
```

Comment:
You should call IotMqtt_Init() before IotMqtt_Cleanup.  Cleanup is meant to undo what Init does, and may act on uninitialized resources if Init was not called beforehand.  (Although neither Init nor Cleanup do much right now, we may change that in the future).

Reply:
Both `IotMqtt_Cleanup` and `IotMqtt_Init` basically update the status of the same global variable `_initCalled`. However, this variable is declared as volatile, so CBMC will always treat it as non-deterministic.
```c
static volatile uint32_t _initCalled = MQTT_LIBRARY_UNINITIALIZED;
```
Therefore, even if we called `IotMqtt_Init` first, it wouldn't make a significant impact in this proof harness. In any case, since the developers may change the implementation of both functions in the future, I added it to the proof. Thanks!

Reply:
I'd add a comment about this to the harness, then, because in your style of writing proof harnesses, this is certainly unusual.

Reply:
Done @markrtuttle